#Primer examen Parcial

Alumno: Camilo Flores Rodríguez

Trabajé solo

#Importar datos
      1. Adquirir datos
      Para adquirir datos descargo con la instrucción del S.O. !wget
      
      He utilizado mi Google drive por lo que debo motar la unidad

In [0]:
!wget https://archive.org/download/nycTaxiTripData2013/trip_data.7z

Observo donde me encuentro ubicado:

In [1]:
ls


gdrive/  sample_data/  trip_data.7z


Puedo observar que sí se descargó en mi unidad drive los archivos que necesito para el ejercicio.

Como ya he montado mi unidad de Google drive me cambio a la carpeta Colab Notebooks

In [2]:
cd gdrive/My\ Drive/'Colab Notebooks'

/content/gdrive/My Drive/Colab Notebooks


In [0]:
Para poder responder la pregunta 3 necesito utilizar la liberría Pandas por lo que la imoporto de una vez

In [0]:
import pandas as pd, time

3.  Una vez descargado el archivo pueden descomprimirlo usando la siguiente instruccion

Continúo con descomprimir un archivo

In [5]:
!7z e trip_data.7z -o/content/gdrive/My\ Drive/Colab\ Notebooks/ExamData trip_data_1.csv


7-Zip [64] 16.02 : Copyright (c) 1999-2016 Igor Pavlov : 2016-05-21
p7zip Version 16.02 (locale=en_US.UTF-8,Utf16=on,HugeFiles=on,64 bits,2 CPUs Intel(R) Xeon(R) CPU @ 2.30GHz (306F0),ASM,AES-NI)

Scanning the drive for archives:
  0M Scan         1 file, 4102781969 bytes (3913 MiB)

Extracting archive: trip_data.7z
--
Path = trip_data.7z
Type = 7z
Physical Size = 4102781969
Headers Size = 433
Method = LZMA:24
Solid = -
Blocks = 12

  0%    
Would you like to replace the existing file:
  Path:     /content/gdrive/My Drive/Colab Notebooks/ExamData/trip_data_1.csv
  Size:     2459600863 bytes (2346 MiB)
  Modified: 2014-05-12 22:02:53
with the file from archive:
  Path:     trip_data_1.csv
  Size:     2459600863 bytes (2346 MiB)
  Modified: 2014-05-12 22:02:53
? (Y)es / (N)o / (A)lways / (S)kip all / A(u)to rename all / (Q)uit? Y

  0% - trip_data_1.csv                        1% - trip_data_1.csv

Busco donde se descomprimió el archivo

In [7]:
ls

'Copia de Copia de SisDist_S1_L1.ipynb'                Examen_Final.ipynb
'Copia de Copy of MCA_SistDist_Ex1'                    Sesion2_Sist_dist2.ipynb
'Copia de Hola, Colaboratory'                          SisDist2_Ex1.ipynb
'Copia de Hola, Colaboratory (1)'                     'SisDist2_S2 (1).ipynb'
'Copia de Hola, Colaboratory (2)'                      SisDist2_S2.ipynb
'Copia de Sesion2_Sist_dist2.ipynb'                    SisDist2_sesion2.ipynb
'Copia de SisDist_S1_L1.ipynb'                         SisDist_S1_L1.ipynb
'Copia de SisDist_S5.ipynb'                            SisDist_S5.ipynb
'Copia de SistDist_S4 (1).ipynb'                       SistDist2/
'Copia de SistDist_S4.ipynb'                           SistDist2_Exam1.ipynb
'Copia de SistDist_S6'                                 SistDist_S4.ipynb
'Copia de Te damos la bienvenida a Colaboratory'       SistDist_S6.ipynb
'Copia de Te damos la bienvenida a Colaboratory (1)'   trip_data.7z
'Copia de Te damos la bienvenida a Cola

El archivo se encuentra en la carpeta que cree ExamData, por lo tanto me cambio de directorio

In [8]:
cd ExamData

/content/gdrive/My Drive/Colab Notebooks/ExamData


In [13]:
ls

trip_data_1.csv*  Untitled0.ipynb


Puedo observar de la celda anterior que el archivo trip_data_1.csv si se encuentra en la carpeta

Leo el archivo con pandas

In [0]:
df = pd.read_csv('trip_data_1.csv')

Verifico si lo leyó bien, para ello visualizo los primeros datos con los nombres de las columnas

In [11]:
df.head()

medallion  ... dropoff_latitude
0  89D227B655E5C82AECF13C3F540D4CF4  ...        40.751171
1  0BD7C8F5BA12B88E0B67BED28BEA73D8  ...        40.750660
2  0BD7C8F5BA12B88E0B67BED28BEA73D8  ...        40.726002
3  DFD2202EE08F7A8DC9A57B02ACB81FE2  ...        40.759388
4  DFD2202EE08F7A8DC9A57B02ACB81FE2  ...        40.747868

[5 rows x 14 columns]

Respondo a las preguntas 

3. Responder las siguientes preguntas:

    1.  Cuantas columnas contiene cada archivo de datos descomprimido
    2.  Cuantos renglones tiene cada archivo
    
    Para ello utilizo shape de pandas y me da la respuesta que busco

In [12]:
df.shape

(14776615, 14)

De acuerdo a la consulta anterior tenemos 14 columnas y 14776615 líneas o registros